nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/main/java/org/apache/nifi/processors/salesforce/QuerySalesforceObject.java

```diff
@@ -182,9 +182,8 @@ public class QuerySalesforceObject extends AbstractProcessor {
     static final PropertyDescriptor AGE_DELAY = new PropertyDescriptor.Builder()
             .name("age-delay")
             .displayName("Age Delay")
-            .description("When 'Age Field' is set the age-based filter will be adjusted by this amount."
-                    + " Only records that are older than the previous run time of this processor, by at least this amount, will be retrieved."
-            )
+            .description("A time window is always between the previous run time of the processor and the current run time. " +
+                    "When 'Age Field' is set, the time window's right edge will be adjusted left by this amount.")
```

Comment:
This initial sentence seems to indicate that the processor will apply a time window regardless of whether the Age Delay property is configured.

Recommend adjusting the wording to indicate the conditional nature of the property. The `left` and `right` wording may be difficult to follow, so it seems better to use something else like `start` and `end`, and include an example.

```suggestion
            .description("When this property is configured, the processor will always apply a time window between the previous run time of the processor and the current run time. " +
                    "The ending timestamp of the time window will be adjusted earlier by the amount configured in this property. For example, with a property value of 10 seconds, an ending timestamp of 12:30:45 would be changed to 12:30:35.")
```

Reply:
I think the first sentence can be omitted.